projects/libsodium/secret_key_auth_fuzzer.cc

```diff
@@ -0,0 +1,22 @@
+extern "C" {
+  #include <sodium.h>
+}
+
+class SodiumState {
+public:
+  unsigned char key[crypto_auth_KEYBYTES];
+  unsigned char mac[crypto_auth_BYTES];
+
+  SodiumState() {
+    sodium_init(); // this can fail with a non-zero return code
```

Comment:
I'll add the assertion - thanks!

I'm not entirely familiar with the conditions necessary to make `sodium_init` fail. I know that it can return 0 if a number of setup conditions fail (detailed in [core.c](https://github.com/jedisct1/libsodium/blob/b1273b04110e19556f7670297ddf447bbd7c95ea/src/libsodium/sodium/core.c#L43)). That said, I've never had it fail to setup.